src/main/java/com/arpnetworking/http/Routes.java

```diff
@@ -318,14 +319,18 @@ public Object apply(final ActorRef telemetry, final boolean isCheck) throws Exce
                 .exceptionally(throwable -> defaultValue);
     }
 
-    private String createMetricName(final HttpRequest request, final String actionPart) {
+    private String createMetricName(final HttpRequest request, final int responseStatus, final String actionPart) {
         final StringBuilder nameBuilder = new StringBuilder()
                 .append(REST_SERVICE_METRIC_ROOT)
                 .append(request.method().value());
-        if (!request.getUri().path().startsWith("/")) {
-            nameBuilder.append("/");
+        if (responseStatus == 404) {
```

Comment:
Pass in the the structure from `response.status()` instead of the integer value and compare it using `.equals` to the literal/enum representing a 404 instead of the magic number.

Reply:
Since the response can be null so it seems we cannot use response.status() as that can not cope the situation when response is null.  (but using integer value can because we assigned responseStatus=599 when response is null).

Reply:
👍